### DISCUSSION TITLE
v2.2.0 is coming (March/April)

### DISCUSSION THREAD

**Discussion by jamietanna** [oapi-codegen/oapi-codegen#1494] [Announcements]
Hey folks, just a head's up that we're targeting late March / early April for the next release of `oapi-codegen`.

My first focus is going to be https://github.com/deepmap/oapi-codegen/pull/1485 to give us a significant revamp in the project's docs, provide a big update to introduce the `CONTRIBUTING.md` and a couple of other things which should improve contribution process.

This release will introduce the ability to use `oapi-codegen` with Go 1.22's enhanced `net/http` routing ([PR](https://github.com/deepmap/oapi-codegen/pull/1475)).

Finally, there are a _tonne_ of changes across the `oapi-codegen` repo, and along the various middlewares, so I'm hoping to get to a good chunk of them.

There's also some stuff happening behind the scenes which _hopefully_ we should be able to talk about before too long 🤞👀

**Comment by jamietanna** [comment]
Forgot to say that I'll be on holiday for the next couple of weeks (celebrating my 30th birthday in Italy) but will get back to things when I'm back

**Reply by GreyXor** [reply]
🎂 🥳 🎉 🎊

**Comment by GreyXor** [comment]
Great news,
That would be possible to have https://github.com/deepmap/oapi-codegen/pull/1477 and https://github.com/oapi-codegen/echo-middleware/pull/15 in 2.2.0 ? i'm waiting for reviews here :) thanks

**Reply by GreyXor** [reply]
@jamietanna hello ? gentle ping

**Reply by jamietanna** [reply]
Thanks, I'm slowly working towards it - with the middleware changes I'm holding them back to try and get to all of them at the same time, to keep them consistent and get them released together. Not forgotten at all

**Reply by GreyXor** [reply]
Thanks :) AMA if needed

**Comment by jamietanna** [comment]
Release checklist:

- [ ] [deepmap/oapi-codegen](https://github.com/deepmap/oapi-codegen)
  - [x] All dependencies updated
  - [ ] Release middleware
  - [ ] Bump middleware (manually)
  - [ ] Release notes ready
- [ ] [oapi-codegen/nethttp-middleware](https://github.com/oapi-codegen/nethttp-middleware)
  - [x] All dependencies updated
  - [x] Release notes ready
- [ ] [oapi-codegen/fiber-middleware](https://github.com/oapi-codegen/fiber-middleware)
  - [x] All dependencies updated
  - [x] Release notes ready
- [ ] [oapi-codegen/iris-middleware](https://github.com/oapi-codegen/iris-middleware)
  - [x] All dependencies updated
  - [x] Release notes ready
- [x] [oapi-codegen/echo-middleware](https://github.com/oapi-codegen/echo-middleware)
  - [x] All dependencies updated
  - [x] Release notes ready
- [x] [oapi-codegen/gin-middleware](https://github.com/oapi-codegen/gin-middleware)
  - [x] All dependencies updated
  - [x] Release notes ready

No changes needed (at least, alongside the main version release):

- [oapi-codegen/runtime](https://github.com/oapi-codegen/runtime)
- [oapi-codegen/nullable](https://github.com/oapi-codegen/nullable)
- [oapi-codegen/testutil](https://github.com/oapi-codegen/testutil)

**Reply by jamietanna** [reply]
Release announce:

```

## Big announcements

In case you've missed it, we've got a few big **??**

- [`oapi-codegen` is moving to its own org (from release v2.3.0)](https://github.com/deepmap/oapi-codegen/discussions/1605)
- [Creating a more sustainable model for `oapi-codegen` in the future](https://github.com/deepmap/oapi-codegen/discussions/1606)

We'd also like to thank our sponsors for this **??**:

TODO
TODO
TODO


**??**

## 🔊 Notable features

The v2.2.0 release was largely focussed on the following big changes, alongside a number of other bug fixes.

### Generate a pure Go 1.22+ `net/http` server

It's now possible to **??**


### Documentation overhaul

A _significant_ undertaking in this release was to rewrite pretty much every line of documentation we had, as well as introducing a number of additional examples and test cases.

We're excited to hear your feedback on the new structure of the README, the additional docs added, and **??**

### JSON Schema configuration

In the spirit of documentation, we now also have a JSON schema **??** which can be further documented in the [Usage section in the README](https://github.com/deepmap/oapi-codegen/blob/v2.2.0/README.md#usage).

This gives you **??**

<!-- Optional: add a release summary here -->
## 🚀 New features and improvements

* feat: add ability to tweak generated names (including use of initialisms) (#1041) @wtertius
* docs: revamp README, add a CONTRIBUTING.md and add a JSON Schema for configuration files (#1485) @jamietanna
* feat: add support for Go 1.22+ net/http routing (#1475) @thedadams

## 🐛 Bug fixes

* Bug: using x-go-type-name results in missing types (#1397) @jamesphillpotts-fr
* fix: correctly register Fiber middleware (#1615) @jamietanna
* fix: ensure we pre-import `nullable` (#1585) @jamietanna
* Prefactor: Changes from #1475 (#1480) @jamietanna
* Ensure docstring comment names are set correctly (#1457) @ConorPKeegan

## 📝 Documentation updates

* docs(examples): document how to set up authenticated middleware with a non-`echo` server (#1627) @jamietanna
* docs(extensions): add missing `x-go-type-name` docs (#1616) @jamietanna
* docs: link to blog post for syncing vendored specs (#1580) @jamietanna
* docs: correct `std-http-server` example (#1576) @sajjanjyothi
* docs: revamp README, add a CONTRIBUTING.md and add a JSON Schema for configuration files (#1485) @jamietanna
* docs: update OpenAPI spec links to `main` branch (#1493) @dharsanb
* Add Fiber example in README (#1495) @alexandregv
* Prefactor: Changes from #1475 (#1480) @jamietanna
* docs: use relative links to files (#1471) @alexandear

## 👻 Maintenance

* chore: add missing test case for replaceInitialism (#1624) @alexandear
* Simplify extensions merge in `mergeOpenapiSchemas` (#1424) @Juneezee
* refactor: extension variable instead of accessing extensions directly (#1448) @alexandear
* chore(deps): remove dependency on `golang.org/x/lint` (#1588) @jamietanna
* docs: revamp README, add a CONTRIBUTING.md and add a JSON Schema for configuration files (#1485) @jamietanna
* chore: add test case for `additionalProperties` bug (#1528) @jamietanna
* Prefactor: Build changes for #1475 (#1482) @jamietanna
* Prefactor: Changes from #1475 (#1480) @jamietanna
* Simplify CI build matrix + build against Go 1.22 (#1459) @jamietanna

## ✍ Other changes

* chore: log warning when using 3.1.x (#1592) @jamietanna

## 📦 Dependency updates

<details>
<summary>16 changes</summary>

* fix(deps): update module golang.org/x/tools to v0.21.0 (#1601) @renovate
* fix(deps): update module github.com/getkin/kin-openapi to v0.124.0 (#1532) @renovate
* chore(deps): update module github.com/golangci/golangci-lint to v1.59.0 (#1625) @renovate
* chore(deps): update module github.com/golangci/golangci-lint to v1.58.2 (#1618) @renovate
* chore(deps): update module github.com/golangci/golangci-lint to v1.58.1 (#1603) @renovate
* fix(deps): update module golang.org/x/text to v0.15.0 (#1591) @renovate
* chore(deps): update module github.com/golangci/golangci-lint to v1.58.0 (#1586) @renovate
* fix(deps): update module golang.org/x/tools to v0.20.0 (#1531) @renovate
* fix(deps): update module golang.org/x/tools to v0.19.0 (#1289) @renovate
* fix(deps): update module github.com/getkin/kin-openapi to v0.123.0 (#1431) @renovate
* chore(deps): update module github.com/golangci/golangci-lint to v1.57.2 (#1466) @renovate
* fix(deps): update module github.com/stretchr/testify to v1.9.0 (#1479) @renovate
* chore(deps): update actions/checkout action to v4 (#1290) @renovate
* chore(deps): update actions/setup-go action to v5 (#1375) @renovate
* Update release-drafter/release-drafter action to v6 (#1453) @renovate
* Update module github.com/golangci/golangci-lint to v1.56.1 (#1455) @renovate
</details>



```

**Reply by jamietanna** [reply]
https://go.dev/ref/mod#go-mod-file-module-deprecation may as well be done too so it's super clear

**Comment by jamietanna** [comment]
And it's, finally, out :rocket: https://github.com/deepmap/oapi-codegen/discussions/1637
